### ISSUE TITLE
[BUG] Issue while animating height

### ISSUE THREAD

**Issue by Ploppy3** [framer/motion#1440]
**Describe the bug**

Hard to describe, I can't figure out what's going on. But there is definitely an issue.

Let me first explain the use case:
I've developed a component that is able to switch between 2 templates, animating smoothly the height between the two.
The animation is as followed:

- Insert the new template initially hidden (opacity 0, height 0)
- Fade out the visible component (opacity 0).
- Simultaneously hide the initial template (height 0) and reveal the new one (height auto)
- Fade in the new template (opacity 1)

Therefore, if considering the two template are identical, the height should be constant as the transition is linear.

https://codesandbox.io/s/twilight-paper-o3xzf?file=/src/Loader.js

While the state of the animation is stable, triggering the transition will make the `<hr />` instantly move downward as if something was taking space in-between the `<hr />` and `<div />` animating above it. There is definitely something: a new `<div />` is been inserted as expected but it has 0 height and overflow hidden, so it shouldn't take any space.

**Steps to reproduce**

On Chome:
Wait for the animation to complete, click the toggle button.
Notice that the `<hr />` moves downward.

**Environment details**

OS: Linux

Does not work on Chrome
Works on Firefox

**Comment by mattgperry** [comment]
You're right that the `hr` moves but I think this is something more to do with CSS and the page layout. Possibly something to do with the margin on `hr` and margin-collapsing when there are multiple elements. But I don't think this is to do with Framer Motion.

**Comment by Ploppy3** [comment]
@mattgperry I don't think this is the case, this is due to unsynchronized animations. Please don't close this issue.
